### PR TITLE
doc: reference: peripherals: Add eSPI interface to peripherals list

### PR DESCRIPTION
**Pull request by albertofloyd**
Add missing eSPI driver interface under peripheral list.

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/reference/peripherals/espi.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+.. _espi_api:
+
+eSPI
+####
+
+Overview
+********
+
```
Comment (gmarull) [inline doc/reference/peripherals/espi.rst:8]: Is it possible to add a short overview? Having an empty section doesn't make much sense

**Review by ceolin — APPROVED** [review]

File: doc/reference/peripherals/espi.rst (hunk 1 above)
Comment (cfriedt) [reply doc/reference/peripherals/espi.rst:8]: Good catch @gmarull.

Please add something minimal in the Overview section @albertofloyd.

**Comment by cfriedt** [comment]
@albertofloyd - could you please add a small bit of text in the Overview section in your PR?

File: doc/reference/peripherals/espi.rst (hunk 1 above)
Comment (albertofloyd) [reply doc/reference/peripherals/espi.rst:18]: Added brief intro on this bus and link to full spec

File: doc/reference/peripherals/espi.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+The eSPI (enhanced serial peripheral interface) is a serial bus that is
+based on SPI. It also features a four-wire interface (receive, transmit, clock
+and slave select) and three configurations: single IO, dual IO and Quad IO.
+
+The technical advancements include lower voltage signal levels (1.8V vs. 3.3V),
+lower pin count, and the frequency is twice as fast (66MHz vs. 33MHz)
+Because of its enhancements, the eSPI is used to replace the LPC
+(lower pin count) interface, SPI, SMBus and sideband signals.
+
+See `eSPI interface specification_` for additional details.
```
Comment (gmarull) [inline doc/reference/peripherals/espi.rst:17]:
```suggestion
See `eSPI interface specification`_ for additional details.
```

File: doc/reference/peripherals/espi.rst [hunk 3]
```diff
@@ -0,0 +1,26 @@
+.. _espi_api:
+
+eSPI
+####
+
+Overview
+********
+The eSPI (enhanced serial peripheral interface) is a serial bus that is
```
Comment (gmarull) [inline doc/reference/peripherals/espi.rst:8]:
```suggestion
********

The eSPI (enhanced serial peripheral interface) is a serial bus that is
```

File: doc/reference/peripherals/espi.rst [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+based on SPI. It also features a four-wire interface (receive, transmit, clock
+and slave select) and three configurations: single IO, dual IO and Quad IO.
```
Comment (gmarull) [inline doc/reference/peripherals/espi.rst:10]:
```suggestion
and slave select) and three configurations: single IO, dual IO and quad IO.
```

**Comment by albertofloyd** [comment]
@gmarull I have applied the suggested changes, let me know if you want me to squash the commits in the PR

**Comment by gmarull** [comment]
> @gmarull I have applied the suggested changes, let me know if you want me to squash the commits in the PR

Yes

**Review by nashif — APPROVED** [review]

**Pull request merged**